### PR TITLE
Add Sentry (and the secret store)

### PR DESCRIPTION
**Pull request by jkmassel**
### Fix
Adds Sentry crash logging (and as a side-effect, support for the secret store).

### Release
These changes are not user-facing, and don't require a release notes update.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/utils/CrashUtils.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    public static void clearCurrentUser() {
+        sharedInstance.currentUser = null;
+        CrashLogging.setNeedsDataRefresh();
+    }
+
+    @Override
+    public String sentryDSN() {
+        return BuildConfig.SENTRY_DSN;
```
Comment (loremattei) [inline Simplenote/src/main/java/com/automattic/simplenote/utils/CrashUtils.java]: This doesn't seem to work well with what we have in the secrets repository. The entry there is "sentryDSN" which would translate to SENTRY_D_S_N here. 
The entry looks correct in the example file here, so I guess you want to change the file in the repo.

**Comment by loremattei** [comment]
The code looks good to me and the secret store integration works well. 
I had a small build issue, as reported above. 
The other issue I have is that, after having fixed the error above locally, my crashes don't show up in Sentry.

File: Simplenote/src/main/java/com/automattic/simplenote/utils/CrashUtils.java (hunk 1 above)
Comment (jkmassel) [reply Simplenote/src/main/java/com/automattic/simplenote/utils/CrashUtils.java]: 🤦‍♂ – you're totally right! Had to rework the BuildConfig logic when merging with `develop` and didn't add it to the store. Updated the PR with the new commit hash and tested this all again from scratch. Should be good now.

Sorry about that!

**Comment by loremattei** [comment]
> Is it possible you have the "Share analytics" switch turned off in Settings? 
It was definitely it! Thanks for the suggestion :-)

**Pull request merged**